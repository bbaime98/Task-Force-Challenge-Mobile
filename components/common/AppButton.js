import React from "react"
import { TouchableOpacity, StyleSheet, Text } from "react-native"
import colors from "../../utils/colors"

const AppButton = ({ title, onPress, style,btnStyle, bgColor = "dark", width = "50%" }) => {
  return (
    <TouchableOpacity
      style={[styles.button, { backgroundColor: colors[bgColor], width }, btnStyle]}
      onPress={onPress}
    >
      <Text style={[styles.text, style]}>{title}</Text>
    </TouchableOpacity>
  )
}

export default AppButton

const styles = StyleSheet.create({
  button: {
    backgroundColor: colors.dark,
    borderRadius: 5,
    justifyContent: "center",
    alignItems: "center",
    padding: 15,
    marginVertical: 35,
  },
  text: {
    color: colors.white,
    fontSize: 15,
    fontFamily: "semiBold",
    textTransform: "uppercase",
  },
})
