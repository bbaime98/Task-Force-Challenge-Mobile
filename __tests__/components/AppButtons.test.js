import React from 'react';
import renderer from 'react-test-renderer';
import AppButton from '../../components/common/AppButton';

jest.useFakeTimers()

it('should render AppButton', () => {
    const rendered = renderer.create(<AppButton />).toJSON();
    expect(rendered).toBeTruthy();
});
it('AppButton snapshot', () => {
    const tree = renderer.create(<AppButton />).toJSON();
    expect(tree).toMatchSnapshot();
});
