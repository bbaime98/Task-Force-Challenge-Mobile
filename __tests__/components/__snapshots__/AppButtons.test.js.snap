// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppButton snapshot 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#0C0D0D",
      "borderRadius": 5,
      "justifyContent": "center",
      "marginVertical": 35,
      "opacity": 1,
      "padding": 15,
      "width": "50%",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontFamily": "semiBold",
          "fontSize": 15,
          "textTransform": "uppercase",
        },
        undefined,
      ]
    }
  />
</View>
`;
